feat(coral): Add activity log

mathieu-anderson opened the pull request:
# Linked issue

Resolves: #2161 #2162

# What kind of change does this PR introduce?

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Docs update
- [ ] CI update

# What is the current behavior?

There is no Activity log

# What is the new behavior?

Add Activity log page 

<img width="1490" alt="Screenshot 2024-01-02 at 17 50 50" src="https://github.com/Aiven-Open/klaw/assets/20607294/fa93eb96-cfa3-42cb-bd89-24029e08c897">


# Other information

In draft because of several things still hanging:
- naming: activity log or audit log?
- `showActivityLog` endpoint:
  - the endpoint does not accept ALL as a value for the env param: should it be added ?
- environment filter: we don't have an endpoint which returns all environments, they are broken up in domains (topic/acl, connectors, schema). This may lead to name conflicts between different domains  (DEV for schema is not DEV for connectors, but they may be called the same). We may want a new endpoint, or we may make several calls and aggregate.  The naming conflict could be fixed with a prefix or something. Like `[Connectors] DEV | [Topics/ ACLs] DEV` **Potential solution**: [e4d0d43](https://github.com/Aiven-Open/klaw/pull/2163/commits/e4d0d4358373973f4386bf2803728ff3cec84bce)
![Screenshot 2024-01-03 at 09 57 00](https://github.com/Aiven-Open/klaw/assets/20607294/12cc6ca4-a6d2-431f-84da-ea993577905e)


- data: sometimes `envName` is missing: 
![Screenshot 2024-01-02 at 17 47 04](https://github.com/Aiven-Open/klaw/assets/20607294/580d0292-1796-4728-adc9-2161a637968a)
![Screenshot 2024-01-02 at 17 49 04](https://github.com/Aiven-Open/klaw/assets/20607294/ab444022-9cbf-4d00-a9fb-95bee39bc2c0)


# Requirements (all must be checked before review)

- [x] The pull request title follows [our guidelines](https://github.com/Aiven-Open/klaw/blob/main/CONTRIBUTING.md#guideline-commit-messages)
- [x] Tests for the changes have been added (if relevant)
- [x] The latest changes from the `main` branch have been pulled
- [x] `pnpm lint` has been run successfully

programmiri commented:
@mathieu-anderson  we should pack that behind a feature flag

mathieu-anderson commented:
@programmiri I didn't add it because I think we use FF to hide incomplete features? This PR is the full activity log feature, if it is merged to `main` it would be usable right away ^^

mathieu-anderson commented:
Huh, ok :o

programmiri commented on coral/src/domain/requests/requests-api.ts:
```diff
@@ -1,20 +1,43 @@
 };
 
-export { getRequestsStatistics, getRequestsWaitingForApproval };
+export { getActivityLog, getRequestsStatistics, getRequestsWaitingForApproval };
```
smol note: `getActivityLog` needs to be exported (and also imported) from the barrel file in `requests/index.ts`, same with types etc that are used outside of `/domain`

programmiri commented on coral/src/domain/environment/environment-api.ts:
```diff
@@ -96,4 +125,5 @@ export {
   getAllEnvironmentsForConnector,
   getPaginatedEnvironmentsForConnector,
   getUpdateEnvStatus,
+  getAllEnvironments,
```
Also needs to be exported from the barrel file.

programmiri reviewed (commented):
Left two small comments, other than that it looks good!

programmiri commented:
>Huh, ok :o 

Yeah, I just feel it's better to be really safe here 🥺 🙈  and be sure we've tested stuff probably before adding it. Although I assume that a lot of people will "only" update Klaw after releases, it's still good to treat `main` as "has to be always release-able" and all.

mathieu-anderson commented:
@programmiri 

- added barrel file for `requests` domain (there was none haha): [4c2e80e](https://github.com/Aiven-Open/klaw/pull/2163/commits/4c2e80eb0c297180a2f70d46f76fd5e5308176f2)
- added feature flag: [c3af986](https://github.com/Aiven-Open/klaw/pull/2163/commits/c3af9865939b7e8a5e963dc2bca1180041bb3152)
